#! /bin/sh /usr/share/dpatch/dpatch-run
## nooptimize.dpatch by Kilian Krause <kilian@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure we respect Debian Policy and only compile -O2
## -- fixed upstream in 1.2 branch

@DPATCH@
diff -urNad asterisk-1.2.0.dfsg~/Makefile asterisk-1.2.0.dfsg/Makefile
--- asterisk-1.2.0.dfsg~/Makefile	2005-11-16 20:23:53.000000000 +0000
+++ asterisk-1.2.0.dfsg/Makefile	2005-11-20 03:45:08.000000000 +0000
@@ -44,7 +44,7 @@
 #K6OPT  = -DK6OPT
 
 #Tell gcc to optimize the code
-OPTIMIZE+=-O6
+OPTIMIZE+=-O2
 endif
 
 #Overwite config files on "make samples"
diff -urNad asterisk-1.2.0.dfsg~/apps/Makefile asterisk-1.2.0.dfsg/apps/Makefile
--- asterisk-1.2.0.dfsg~/apps/Makefile	2005-11-11 00:32:45.000000000 +0000
+++ asterisk-1.2.0.dfsg/apps/Makefile	2005-11-20 03:42:28.000000000 +0000
@@ -112,7 +112,7 @@
 	$(CC) $(SOLINK) -o $@ ${CYGSOLINK} $< ${CYGSOLIB} -lodbc
 
 look:	look.c
-	$(CC) -pipe -O6 -g look.c -o look -lncurses
+	$(CC) -pipe -O2 -g look.c -o look -lncurses
 
 ifeq (SunOS,$(shell uname))
 app_chanspy.so: app_chanspy.o
diff -urNad asterisk-1.2.0.dfsg~/cdr/Makefile asterisk-1.2.0.dfsg/cdr/Makefile
--- asterisk-1.2.0.dfsg~/cdr/Makefile	2005-11-14 00:45:07.000000000 +0000
+++ asterisk-1.2.0.dfsg/cdr/Makefile	2005-11-20 03:44:08.000000000 +0000
@@ -32,7 +32,7 @@
 #the default as we now have a better instruction set to work with. - Belgarath
 ifeq ($(PROC),sparc64)
  PROC=ultrasparc
-  CFLAGS += -mtune=$(PROC) -O3 -pipe -fomit-frame-pointer -mcpu=v8
+  CFLAGS += -mtune=$(PROC) -O2 -pipe -fomit-frame-pointer -mcpu=v8
 endif
 
 #
diff -urNad asterisk-1.2.0.dfsg~/channels/Makefile asterisk-1.2.0.dfsg/channels/Makefile
--- asterisk-1.2.0.dfsg~/channels/Makefile	2005-11-17 04:26:07.000000000 +0000
+++ asterisk-1.2.0.dfsg/channels/Makefile	2005-11-20 03:44:19.000000000 +0000
@@ -44,7 +44,7 @@
 
 ifeq ($(PROC),sparc64)
   PROC=ultrasparc
-  CFLAGS += -mtune=$(PROC) -O3 -pipe -fomit-frame-pointer -mcpu=v8
+  CFLAGS += -mtune=$(PROC) -O2 -pipe -fomit-frame-pointer -mcpu=v8
 endif
 
 ifeq (${OSARCH},FreeBSD)
diff -urNad asterisk-1.2.0.dfsg~/codecs/gsm/Makefile asterisk-1.2.0.dfsg/codecs/gsm/Makefile
--- asterisk-1.2.0.dfsg~/codecs/gsm/Makefile	2005-11-08 03:31:45.000000000 +0000
+++ asterisk-1.2.0.dfsg/codecs/gsm/Makefile	2005-11-20 03:44:25.000000000 +0000
@@ -69,7 +69,7 @@
 #So we go lowest common available by gcc and go a step down, still a step up from
 #the default as we now have a better instruction set to work with. - Belgarath
 ifeq (${PROC},ultrasparc)
-OPTIMIZE+=-mcpu=v8 -mtune=$(PROC) -O3 
+OPTIMIZE+=-mcpu=v8 -mtune=$(PROC) -O2 
 endif
 
 PG =
diff -urNad asterisk-1.2.0.dfsg~/codecs/ilbc/Makefile asterisk-1.2.0.dfsg/codecs/ilbc/Makefile
--- asterisk-1.2.0.dfsg~/codecs/ilbc/Makefile	2005-11-08 04:13:18.000000000 +0000
+++ asterisk-1.2.0.dfsg/codecs/ilbc/Makefile	2005-11-20 03:44:30.000000000 +0000
@@ -1,5 +1,5 @@
 ARCH=$(PROC)
-CFLAGS+=-Wall -O3 -funroll-loops
+CFLAGS+=-Wall -O2 -funroll-loops
 ifneq (${OSARCH},CYGWIN)
 CFLAGS += -fPIC
 endif
diff -urNad asterisk-1.2.0.dfsg~/codecs/lpc10/Makefile asterisk-1.2.0.dfsg/codecs/lpc10/Makefile
--- asterisk-1.2.0.dfsg~/codecs/lpc10/Makefile	2005-11-08 04:13:19.000000000 +0000
+++ asterisk-1.2.0.dfsg/codecs/lpc10/Makefile	2005-11-20 03:44:50.000000000 +0000
@@ -40,7 +40,7 @@
 #So we go lowest common available by gcc and go a step down, still a step up from
 #the default as we now have a better instruction set to work with. - Belgarath
 ifeq ($(PROC),ultrasparc)
-        CFLAGS+= -mtune=$(PROC) -mcpu=v8 -O3 -fomit-frame-pointer
+        CFLAGS+= -mtune=$(PROC) -mcpu=v8 -O2 -fomit-frame-pointer
 else
 ifneq ($(OSARCH),SunOS)
 ifneq  ($(OSARCH),arm)
diff -urNad asterisk-1.2.0.dfsg~/db1-ast/Makefile asterisk-1.2.0.dfsg/db1-ast/Makefile
--- asterisk-1.2.0.dfsg~/db1-ast/Makefile	2004-12-14 23:36:30.000000000 +0000
+++ asterisk-1.2.0.dfsg/db1-ast/Makefile	2005-11-20 03:44:58.000000000 +0000
@@ -10,7 +10,7 @@
 #Added support for UltraSparc - Belgarath
 ifeq ($(ARCH),sparc64)
 PROC=ultrasparc
-CFLAGS += -mtune=$(PROC) -O3 -pipe -fomit-frame-pointer -mcpu=v8
+CFLAGS += -mtune=$(PROC) -O2 -pipe -fomit-frame-pointer -mcpu=v8
 endif
 
 LIBDBSO=libdb.so.$(SOVER)
